pkg/store/store.go

```diff
@@ -36,11 +36,10 @@ type Store interface {
 	// Cancel watch key
 	CancelWatch(key string) error
 
-	// Acquire the lock at key
-	Acquire(key string, value []byte) (string, error)
-
-	// Release the lock at key
-	Release(session string) error
+	// CreateLock for a given key.
+	// The returned Locker is not held and must be acquired with `.Lock`.
+	// value is optional.
+	CreateLock(key string, value []byte) (Locker, error)
```

Comment:
```go
// Anyone can create the same lock even if the lock is held
l1, _ := store.CreateLock("/foo)
l1.Lock()
l2, _ := store.CreateLock("/foo")
l3, _ := store.CreateLock("/foo")
l2.Lock() // this will block until l1 is released
```